lib/ansible/module_utils/common/parameters.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.common._collections_compat import (
+    KeysView,
+    Sequence,
+)
+
+from ansible.module_utils.six import (
+    binary_type,
+    text_type,
+)
+
+PASS_VARS = {
+    'check_mode': 'check_mode',
+    'debug': '_debug',
+    'diff': '_diff',
+    'keep_remote_files': '_keep_remote_files',
+    'module_name': '_name',
+    'no_log': 'no_log',
+    'remote_tmp': '_remote_tmp',
+    'selinux_special_fs': '_selinux_special_fs',
+    'shell_executable': '_shell',
+    'socket': '_socket_path',
+    'syslog_facility': '_syslog_facility',
+    'tmpdir': '_tmpdir',
+    'verbosity': '_verbosity',
+    'version': 'ansible_version',
+}
+
+# Note: When getting Sequence from collections, it matches with strings. If
+# this matters, make sure to check for strings before checking for sequencetype
+SEQUENCETYPE = frozenset, KeysView, Sequence
+
+
+def handle_aliases(params, argument_spec, legal_inputs=None, spec=None, param=None):
+    if legal_inputs is None:
+        legal_inputs = ['_ansible_%s' % k for k in PASS_VARS]
```

Comment:
We should probably make this a python module-level constant.

Reply:
That way we don't have to set it every time we call this function.

Reply:
Hm.... I just noticed that legal_inputs is being modified below as well.  That's not great practice unless we know that we could be dealing with really large lists.  That's not the case here so I think we should do something like:
``` python
# Module global scope
DEFAULT_LEGAL_PARAMS = ['_ansible_%s' % k for k in PASS_VARS]
[...]
def handle_aliases([...]):
    if legal_inputs is None:
        legal_inputs = DEFAULT_LEGAL_PARAMS[:]
```